Python/Product/VSInterpreters/PipPackageManager.cs

```diff
@@ -675,37 +675,50 @@ private sealed class Suppressed : IDisposable {
             }
         }
 
-        private void RefreshIsCurrentTimer_Elapsed(object state) {
+        private async void RefreshIsCurrentTimer_Elapsed(object state) {
             if (_isDisposed) {
                 return;
             }
 
+            try {
+                _refreshIsCurrentTrigger.Change(Timeout.Infinite, Timeout.Infinite);
+            } catch (ObjectDisposedException) {
+            }
+
             InstalledFilesChanged?.Invoke(this, EventArgs.Empty);
 
             var cts = new CancellationTokenSource();
             var cancellationToken = cts.Token;
             var oldCts = Interlocked.Exchange(ref _currentRefresh, cts);
             try {
                 oldCts?.Cancel();
+                oldCts?.Dispose();
             } catch (ObjectDisposedException) {
             }
-            oldCts?.Dispose();
 
-            CacheInstalledPackagesAsync(false, false, cancellationToken)
-                .SilenceException<OperationCanceledException>()
-                .DoNotWait();
+            try {
+                await CacheInstalledPackagesAsync(false, false, cancellationToken);
+            } catch (OperationCanceledException) {
+            } catch (Exception ex) when (!ex.IsCriticalException()) {
+                Debug.Fail(ex.ToUnhandledExceptionMessage(GetType()));
+            }
         }
 
         public void NotifyPackagesChanged() {
             _refreshIsCurrentTrigger.Change(100, Timeout.Infinite);
         }
 
         private void OnRenamed(object sender, RenamedEventArgs e) {
-            _refreshIsCurrentTrigger.Change(1000, Timeout.Infinite);
+            if (ModulePath.IsPythonFile(e.FullPath, false, true, false) ||
```

Comment:
I believe we'll get a notification for each file contained in the folder, but I'll check.